vaadinfrontend/src/main/java/life/qbic/datamanager/views/project/view/components/ProjectLinksComponent.java

```diff
@@ -48,30 +57,69 @@ public ProjectLinksComponent(@Autowired ProjectInformationService projectInforma
           button.addThemeVariants(ButtonVariant.LUMO_ICON,
               ButtonVariant.LUMO_ERROR,
               ButtonVariant.LUMO_TERTIARY);
-          button.addClickListener(e -> {
-            linkList.remove(projectLink);
-            projectLinks.getDataProvider().refreshAll();
-            }
-          );
-          button.setIcon(new Icon("lumo","cross"));
+          button.setIcon(new Icon("lumo", "cross"));
+          button.addClickListener(e -> removeLink(projectLink));
         })
     );
-    projectLinks.setItems(linkList);
+    projectLinks.setItems(new ArrayList<>());
+    offerSearchComponent.addSelectedOfferChangeListener(it -> {
+      if (Objects.isNull(it.getValue())) {
+        return;
+      }
+      if (!it.isFromClient()) {
+        return;
+      }
+      if (it.getValue() != it.getOldValue()) {
+        addLink(offerLink(it.getValue().offerId()));
+        it.getSource().clearSelection();
+      }
+    });
 
     getContent().addTitle("Links");
-    getContent().addFields(projectLinks);
+    getContent().addFields(offerSearchComponent, projectLinks);
+  }
+
+  private static ProjectLink offerLink(OfferIdentifier offerIdentifier) {
+    return ProjectLink.of(OFFER_TYPE_NAME, offerIdentifier.value());
+  }
+
+  private static ProjectLink offerLink(OfferId offerIdentifier) {
+    return ProjectLink.of(OFFER_TYPE_NAME, offerIdentifier.id());
+  }
```

Comment:
I'm confused why we have OfferId and OfferIdentifier with an interchangeable meaning in this context 😕

Reply:
We have both because we get them from different contexts. One is obtained by reading linked offers in a project, the other is obtained by searching offers based on some property.
I will have another look